input:  vmu_rt1170: build failure for input related cases

**Describe the bug**

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
input.api.thread | v3.6.0-2137-gcd2c7874b95e | Build failure | vmu_rt1170/mimxrt1176/cm7 | tests/subsys/input/api
input.api.synchronous | v3.6.0-2137-gcd2c7874b95e | Build failure | vmu_rt1170/mimxrt1176/cm7 | tests/subsys/input/api
sample.usb.hid-mouse | v3.6.0-2137-gcd2c7874b95e | Build failure | vmu_rt1170/mimxrt1176/cm7 | samples/subsys/usb/hid-mouse
sample.input.input_dump | v3.6.0-2137-gcd2c7874b95e | Build failure | vmu_rt1170/mimxrt1176/cm7 | samples/subsys/input/input_dump
sample.input.input_dump_shell | v3.6.0-2137-gcd2c7874b95e | Build failure | vmu_rt1170/mimxrt1176/cm7 | samples/subsys/input/input_dump

</body>
</html>

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - vmu_rt1170
 - What have you tried to diagnose or workaround this issue?
 - 
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
ubuntu@ubuntu-OptiPlex-7050:/home/shared/disk/zephyr_project/zephyr_test/zephyr$ git bisect bad
c032c6305dccaa74c6095fc27f382aaa7ce2730f is the first bad commit
commit c032c6305dccaa74c6095fc27f382aaa7ce2730f
Author: Benjamin Perseghetti <bperseghetti@rudislabs.com>
Date:   Tue Apr 2 20:26:46 2024 +0200

    boards: nxp: vmu_rt1170: Fix dts binding issues
    
    Fixs USDHC by setting PWR and CD gpio's correctly.
    Adds missing button from the GPS module.
    
    Co-authored-by: Peter van der Perk <peter.vanderperk@nxp.com>
    Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

 boards/nxp/vmu_rt1170/vmu_rt1170-pinctrl.dtsi | 13 +++++++++++--
 boards/nxp/vmu_rt1170/vmu_rt1170.dts          | 17 ++++++++++-------
 2 files changed, 21 insertions(+), 9 deletions(-)

```


**To Reproduce**
Steps to reproduce the behavior:
scripts/twister -b vmu_rt1170 -T vmu_rt1170/mimxrt1176/cm7

**Expected behavior**
pass

**Impact**
all input muldules for vmu_rt1170

**Logs and console output**

```
zephyr/drivers/input/CMakeFiles/drivers__input.dir/input_gpio_keys.c.obj.d -o zephyr/drivers/input/CMakeFiles/drivers__input.dir/input_gpio_keys.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/input/input_gpio_keys.c
In file included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/irq_multilevel.h:15,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:20,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/device.h:12,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/input/input_gpio_keys.c:9:
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: "zephyr-code must be specified to use the input-gpio-keys driver"
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                    ^~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used:  v3.6.0-2137-gcd2c7874b95e

--- a/boards/nxp/vmu_rt1170/vmu_rt1170.dts
+++ b/boards/nxp/vmu_rt1170/vmu_rt1170.dts
@@ -9,6 +9,7 @@
 #include <nxp/nxp_rt11xx_cm7.dtsi>
 #include <zephyr/dt-bindings/led/led.h>
 #include "vmu_rt1170.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
 
 / {
 	model = "NXP VMU RT1170";
@@ -35,12 +36,13 @@
 		zephyr,code-partition = &slot0_partition;
 	};
 
-	/* This is the Button on the included GPS module for 10 pin JST-GH */
-	buttons {
+	/* This is the Arming Button on the included GPS module for 10 pin JST-GH */
+	gpio_keys {
 		compatible = "gpio-keys";
 		arming_button: button_0 {
 			label = "Arming Switch";
 			gpios = <&gpio1 24 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+			zephyr,code = <INPUT_KEY_0>;
 		};
 	};
 

